src/property.ts

```diff
@@ -925,8 +925,9 @@ function Property$ensureInited(property: Property, obj: Entity): void {
 		if (!property.isCalculated) {
 			Property$init(property, obj, Property$getInitialValue(property));
 
-			// Mark the property as pending initialization if it still has no underlying value to allow default calculation rules to run for it
-			if (obj.__fields__[property.name] === null)
+			const underlyingValue = obj.__fields__[property.name];
+			// Mark the property as pending initialization if it still has no underlying value, or is the property type's default, to allow default calculation rules to run for it
+			if (underlyingValue === property.defaultValue || (property.isList && underlyingValue.length === 0))
```

Comment:
Do you think it would be worth sanity checking for an array? List properties by default would be initialized as a new array, but maybe there is the possibility of someone setting their own default which could be null/undefined/junk?